docs/shared/tutorials/epic-web-app.md

```diff
@@ -0,0 +1,495 @@
+      "cache": true
+    },
+    "lint": {
+      "cache": true
+    },
+    "coverage": {
+      "cache": true
+    },
+    "typecheck": {
+      "cache": true
+    }
+  },
+  "defaultBase": "main"
+}
+```
+
+Try running `typecheck` a second time:
+
+```shell {% path="~/epic-web-app" %}
+npx nx typecheck
+```
+
+The first time `nx typecheck` was run, it took about 4 seconds - just like running `npm run typecheck`. But the second time you run `nx typecheck`, it completes instantly and displays this message:
+
+```text
+Nx read the output from the cache instead of running the command for 1 out of 1 tasks.
+```
+
+You can see the same caching behavior working when you run `npx nx lint` or `npx nx test:e2e:run`.
+
+```shell {% path="~/epic-web-app" %}
+npx nx lint
+npx nx test:e2e:run
+```
+
+The initialization script also added `nx exec --` to all the `package.json` scripts that are cacheable. This allows Nx to cache those scripts even if you execute them using `npm run typecheck` or `npm run lint`.
+
+## Use Nx Plugins to Enhance Playwright Tasks with Caching
+
+You may have noticed that `test:e2e:run` created a `playwright-report` folder and a `test-results` folder. But we only specified `playwright-report` in the `outputs` property. Let's examine why this is a problem. Delete the `playwright-report` and `test-results` folders and then re-run `test:e2e:run`.
+
+```shell {% path="~/epic-web-app" %}
+npx nx test:e2e:run
+```
+
+This replays the task from the cache and it recreates the `playwright-report` folder, but it doesn't recreate the `test-results` folder. We can fix this by updating the settings in the `nx.json` file.
+
+```json {% fileName="nx.json" %}
+{
+  "targetDefaults": {
+    // ...
+    "test:e2e:run": {
+      "outputs": [
+        "{projectRoot}/playwright-report",
+        "{projectRoot}/test-results"
+      ],
+      "cache": true
+    }
+    // ...
+  }
+}
+```
+
+Now you can delete the `playwright-report` and `test-results` folders and re-run `test:e2e:run` again. This time, both folders will be recreated.
+
+```shell {% path="~/epic-web-app" %}
+npx nx test:e2e:run
+```
+
+There is still a problem with this set up. The `outputs` property is currently hard-coded in the `nx.json` file. If someone were to set the `outputDir` property in `playwright.config.ts` file, they would need to also update the `outputs` property for `test:run:e2e` to match. This is where plugins can help. They directly infer information from the actual tooling configuration files (`playwright.config.ts` in this case).
+
+Nx plugins can:
+
+- automatically configure caching for you, including inputs and outputs based on the underlying tooling configuration
+- infer tasks that can be run on a project because of the tooling present
+- provide code generators to help scaffold out projects
+- automatically keep the tooling versions and configuration files up to date
+
+For this tutorial, we'll just focus on the automatic caching configuration.
```

Comment:
should we mention also the task splitting feature though? Maybe just as a small hint as clearly the full benefit would emerge in combination with Nx Agents on CI.